### ISSUE TITLE
Large CPU consumption

### ISSUE THREAD

**Issue by mdevaev** [pi-kvm/ustreamer#3]
Thanks Maxim, great work, latest master working here:

``` 
./stream.sh                                                                                                                                              
-- INFO  [101283.243 tid=16956] -- Installing SIGINT handler ...
-- INFO  [101283.243 tid=16956] -- Installing SIGTERM handler ...
-- INFO  [101283.243 tid=16956] -- Ignoring SIGPIPE ...
-- INFO  [101283.243 tid=16956] -- Using JPEG quality: 100%
-- INFO  [101283.243 tid=16956] -- Listening HTTP on [0.0.0.0]:18000
-- INFO  [101283.243 tid=16957] -- Using V4L2 device: /dev/video0
-- INFO  [101283.243 tid=16957] -- Using desired FPS: 30
================================================================================
-- INFO  [101283.243 tid=16958] -- Starting HTTP eventloop ...
-- INFO  [101283.244 tid=16957] -- Device fd=8 opened
-- INFO  [101283.244 tid=16957] -- Using input channel: 0
-- INFO  [101283.244 tid=16957] -- Using TV standard: DEFAULT
-- INFO  [101283.244 tid=16957] -- Using resolution: 1920x1080
-- INFO  [101283.244 tid=16957] -- Using pixelformat: MJPG
-- INFO  [101283.264 tid=16957] -- Requested 17 HW buffers, got 17
-- INFO  [101283.340 tid=16957] -- Capturing started
-- INFO  [101283.340 tid=16957] -- Switching to HW encoder because the input format is (M)JPEG
-- ERROR [101283.340 tid=16957] -- Can't query HW JPEG compressor params and set quality (unsupported)
-- INFO  [101283.340 tid=16957] -- Spawning 17 workers ...
-- INFO  [101283.340 tid=16957] -- Capturing ...
```

It seems ustreamer consumes a little more CPU than mjpg-streamer does with similar settings, is there maybe any kind of re-encoding happening or any other idea why?

![screenshot_2019-03-03_13-15-43](https://user-images.githubusercontent.com/3270028/53695724-098c2600-3db7-11e9-825d-56414f192ee3.png)
![screenshot_2019-03-03_13-19-32](https://user-images.githubusercontent.com/3270028/53695725-0d1fad00-3db7-11e9-81fa-2c8373d83b3c.png)

I'll wrap a small ebuild around it to make it easy for gentoo users to use as well.

_Originally posted by @chron0 in https://github.com/pi-kvm/ustreamer/issues/2#issuecomment-469022774_

**Comment by mdevaev** [comment]
TL;DR from https://github.com/pi-kvm/ustreamer/issues/2

> One of the differences with mjpg-streamer is that ustreamer always reads the data from the camera, regardless of whether there are any clients who are looking at the stream. The main reason is that the usteramer was designed to use with a separate broadcast control daemon (https://github.com/pi-kvm/kvmd, https://github.com/pi-kvm/kvmd/blob/master/kvmd/apps/kvmd/streamer.py). I assumed that if there were clients, this would launch the ustreamer; after disconnecting the last one, it would kill the process. Therefore, in an inactive state, my program consumes more cpu time. However, if there are clients, the result should not differ significantly from mjpg-streamer.

https://github.com/pi-kvm/ustreamer/issues/2#issuecomment-469026952
